### DISCUSSION TITLE
[LAMMPS] Potential Comparison

### DISCUSSION THREAD

**Discussion by AnuragKr** [deepmodeling/deepmd-kit#2133] [Q&A] [answered]
Hi @njzjz ,

I am trying to analyze how existing potential performance as compared to DeePMD potential.

LAMMPS Code For Copper with DeePMD potential --

```lammps
units           metal
dimension       3
boundary        p p p
atom_style      atomic

variable latparam equal 3.61
variable material string Cu

lattice         fcc ${latparam}
region          box block 0 ${xx} 0 ${yy} 0 ${zz}
create_box      1 box
create_atoms    1 random 100000 123456 box
mass            1 63.546

reset_timestep  0
timestep 0.0005
velocity        all create 300 87287 loop geom

neighbor        1.0 bin
neigh_modify    delay 5 every 1

pair_style deepmd copper_single_gpu_graph-compress.pb
pair_coeff * *


neighbor        0.3 bin
neigh_modify    delay 0 every 20 check no

fix             1  all  nve

# Set thermo output
thermo 100
thermo_style custom step lx ly lz press pxx pyy pzz pe temp

run             10000
                         
```
Output--

![Cu_1lac_updated](https://user-images.githubusercontent.com/10074710/204039604-75feb921-bd49-4682-ba0f-333e47b45572.png)

LAMMPS Code For Copper with EAM potential --

```lammps
units           metal
dimension       3
boundary        p p p
atom_style      atomic

variable latparam equal 3.61
variable material string Cu

lattice         fcc ${latparam}
region          box block 0 ${xx} 0 ${yy} 0 ${zz}
create_box      1 box
create_atoms    1 random 100000 123456 box
mass            1 63.546

reset_timestep  0
timestep 0.0005
velocity        all create 300 87287 loop geom

neighbor        1.0 bin
neigh_modify    delay 5 every 1

pair_style eam
pair_coeff * * Cu_u3.eam


neighbor        0.3 bin
neigh_modify    delay 0 every 20 check no

fix             1 all nve

# Set thermo output
thermo 100
thermo_style custom step lx ly lz press pxx pyy pzz pe temp

run             10000
```
Output --

![Cu_1lac_eam](https://user-images.githubusercontent.com/10074710/204039651-de16ea83-33ba-4812-80b3-be63307ec55c.png)

Questions--
1) Is this the correct way of comparing?
2) What metric should I consider for comparison?
3) If the wall time of EAM coming less than DeePMD potential then can we say It is better?

System -- 2 GPU Turing Architecture

**Comment by wanghan-iapcm** [comment, marked as answer]
If you want a fair comparison on the efficiency, you may want to use the same system (same atom configuration and cell size), the same computational resources and run under the same MD simulation protocol. 

Please note that merely a higher efficiency does not mean the EAM potential is better than DP. The accuracy is always taken into consider. In general DP is slower than EAM, but it is usually more accurate. Therefore one need to consider both accuracy and efficiency when he/she chooses a potential model.

**Reply by AnuragKr** [reply]
Thanks, @wanghan-iapcm  for your response. Your response is always very elaborate. Just a few more doubts -
1. In the above LAMMPs code Is there any dissimilarity for comparison? I have used the same atom configuration but the pair style is different as it varies based on potential. And also cell size I have kept the same.
2. For efficiency we need to evaluate based on wall time, timestep/sec, etc., and for accuracy energy, force, and virial tensor.

**Reply by wanghan-iapcm** [reply]
From the screen shots, there is a difference on the number of total and peratom number of neighbors.

**Reply by AnuragKr** [reply]
Please suggest how I can make sure the exact number of neighbor atoms is processed. It's coming differently every time.

**Reply by wanghan-iapcm** [reply]
Please check:
1. the configuration is the same
2. the cut-off radius of the potentials are the same
3. the shell/bin sizes used when building the neighbor list are the same
